src/main/java/it/gov/pagopa/payhub/auth/service/user/IamUserInfoDTO2UserInfoMapper.java

```diff
@@ -27,26 +29,48 @@ public IamUserInfoDTO2UserInfoMapper(UsersRepository usersRepository, OperatorsR
 
     @Override
     public UserInfo apply(IamUserInfoDTO iamUserInfoDTO) {
+        if (iamUserInfoDTO.isSystemUser()) {
+            return systemUserMapper(iamUserInfoDTO);
+        }
+        return userInfoMapper(iamUserInfoDTO);
+    }
+
+    private UserInfo systemUserMapper(IamUserInfoDTO iamUserInfoDTO) {
+        return UserInfo.builder()
+          .userId(iamUserInfoDTO.getUserId())
+          .mappedExternalUserId(iamUserInfoDTO.getFiscalCode())
```

Comment:
@antocalo could it be better if this could be `CodIPA + "WS_USER"`? this way it should be already handled by MyPay code

Reply:
Considering that some system users are saved as CodIpa-WS_USER, I propose to maintain this naming convention. Just pay attention to the separator "-" between codipa and WS_USER.